### PR TITLE
RSE-1196: Add Case Category Instance Field To Case Category Creation Form

### PR DESCRIPTION
**Pull request by tunbola**
## Overview
This PR adds the Case Category Instance Type form field to the case category creation form. The field allows users to select from a dropdown list of available instance types.

## Before
The Instance type form field was not available.

## After
The Instance type form field was added.

Instance Type

## Comments
A couple of fixes were also done in this PR:
- The ID column which was omitted in an earlier PR was added to the case_category_instance table
- The default instance type was renamed to Case Management
- Errors in the case category instance API file was fixed.

### CODE REVIEW

File: CRM/Civicase/DAO/CaseCategoryInstance.php [hunk 1]
```diff
@@ -1,14 +1,12 @@
 <?php
 
 /**
- * Civicrm DAO File.
- *
```
Comment (ivan-compucorp) [inline CRM/Civicase/DAO/CaseCategoryInstance.php:5]: Why is this removed? Maybe it's better to write some one-line description here?

File: sql/auto_install.sql [hunk 2]
```diff
@@ -23,7 +23,9 @@ CREATE TABLE IF NOT EXISTS `civicase_contactlock` (
 -- *
 -- *******************************************************/
 CREATE TABLE IF NOT EXISTS `civicrm_case_category_instance` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT  COMMENT 'Unique CaseCategoryInstance Id',
   `category_id` int unsigned NOT NULL   COMMENT 'One of the values of the case_type_categories option group',
   `instance_id` int unsigned NOT NULL   COMMENT 'One of the values of the case_category_instance_type option group',
+  PRIMARY KEY (`id`),
   UNIQUE INDEX `unique_category`(category_id)
  ) ENGINE=InnoDB DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci;
```
Comment (ivan-compucorp) [inline sql/auto_install.sql:31]: Should we add new line at the end of the file?

File: templates/CRM/Civicase/Form/CaseCategoryInstance.tpl [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    });
+    {/literal}
+</script>
+
+<table>
+  <tr id="case_category_instance_type">
+    <td class="label"> {$form.case_category_instance_type.label} </td>
+    <td> {$form.case_category_instance_type.html} </td>
+  </tr>
+</table>
```
Comment (ivan-compucorp) [inline templates/CRM/Civicase/Form/CaseCategoryInstance.tpl:14]: Should we add new line at the end of the file?

File: CRM/Civicase/DAO/CaseCategoryInstance.php (hunk 1 above)
Comment (tunbola) [reply CRM/Civicase/DAO/CaseCategoryInstance.php:5]: @ivan-compucorp This is autogenerated by Civix.
